libbeat/processors/actions/decode_json_fields.go

```diff
@@ -75,7 +77,12 @@ func (f decodeJSONFields) Run(event common.MapStr) (common.MapStr, error) {
 				continue
 			}
 
-			_, err = event.Put(field, output)
+			if f.target != nil {
```

Comment:
There will be three cases to handle here. When `*f.target == ""` you can't use `event.Put`, instead you need to add the fields from `output` to `event`. Looking at the json decoding in Filebeat, the default behavior is to not overwrite keys unless [`overwrite_key`](https://www.elastic.co/guide/en/beats/filebeat/5.0/configuration-filebeat-options.html#config-json) is set to true. So I would follow this model to be consistent.